src/ChanperfTracker.ts

```diff
@@ -0,0 +1,156 @@
+                this.chanperfHistory.userCpuUsage = parseInt(userCpuUsage) > 100 ? 100 : parseInt(userCpuUsage);
+                this.chanperfHistory.sysCpuUsage = parseInt(sysCpuUsage) > 100 ? 100 : parseInt(sysCpuUsage);
+
+                this.chanperfHistory.dataSet.totalMemKib.push(parseInt(totalMemKib));
+                this.chanperfHistory.dataSet.anonMemKib.push(parseInt(anonMemKib));
+                this.chanperfHistory.dataSet.fileMemKib.push(parseInt(fileMemKib));
+                this.chanperfHistory.dataSet.sharedMemKib.push(parseInt(sharedMemKib));
+                this.chanperfHistory.dataSet.totalCpuUsage.push(parseInt(totalCpuUsage) > 100 ? 100 : parseInt(totalCpuUsage));
+                this.chanperfHistory.dataSet.userCpuUsage.push(parseInt(userCpuUsage) > 100 ? 100 : parseInt(userCpuUsage));
+                this.chanperfHistory.dataSet.sysCpuUsage.push(parseInt(sysCpuUsage) > 100 ? 100 : parseInt(sysCpuUsage));
+
+                if (!this.filterOutLogs) {
+                    normalOutput += line + '\n';
+                }
+                dataChanged = true;
+            } else {
+                // see the following for an explanation for this regex: https://regex101.com/r/Nwqd5e/1/
+                let noInfoAvailableMatch = /channel: *(mem *and *cpu *data *not *available)/gim.exec(line);
+
+                if (noInfoAvailableMatch) {
+                    this.chanperfHistory.missingInfoMessage = noInfoAvailableMatch[1];
+
+                    if (!this.filterOutLogs) {
+                        normalOutput += line + '\n';
+                    }
+                    dataChanged = true;
+                } else if (line) {
+                    normalOutput += line + '\n';
+                }
+            }
+
+        }
+
+        if (dataChanged) {
+            this.emit('chanperf-event', this.chanperfHistory);
+        }
+
+        return normalOutput;
+    }
+
+    /**
+     * Helper function used to create a new ChanperfHistory object with default values
+     */
+    private createNewChanperfHistory(): ChanperfHistory {
+        return {
+            totalMemKib: 0,
+            anonMemKib: 0,
+            fileMemKib: 0,
+            sharedMemKib: 0,
+            totalCpuUsage: 0,
+            userCpuUsage: 0,
+            sysCpuUsage: 0,
+            dataSet: {
+                totalMemKib: [],
+                anonMemKib: [],
+                fileMemKib: [],
+                sharedMemKib: [],
+                totalCpuUsage: [],
+                userCpuUsage: [],
+                sysCpuUsage: []
+            }
+        };
+    }
+}
+
+export interface ChanperfHistory {
+    missingInfoMessage?: string;
+    totalMemKib: number;
+    anonMemKib: number;
+    fileMemKib: number;
+    sharedMemKib: number;
+    totalCpuUsage: number;
+    userCpuUsage: number;
+    sysCpuUsage: number;
+    dataSet: ChanperfDataSet;
+}
+
+interface ChanperfDataSet {
```

Comment:
I don't think we need this. An array of `ChanperfHistory` objects is enough for any charting library to be able to generate the datasets it needs.

Reply:
So I was thinking about this. I think you are right that we don't need to send the entire history back BUT we should send history back for every match in the supplied string. Since we are passing in and support a multi line string we should still parse out each event that was in that string. The main thing is that we don't need to store the history in a persistent way since the client could do that.